client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,191 @@
-jest.mock( 'calypso/blocks/dismissible-card', () => {
-	return function DismissibleCard() {
-		return null;
-	};
-} );
-
-jest.mock( 'calypso/lib/analytics/track-component-view', () => {
-	return function TrackComponentView() {
-		return null;
-	};
-} );
-
-import { Card, Button } from '@automattic/components';
-import { shallow } from 'enzyme';
-import PlanPrice from 'calypso/my-sites/plan-price/';
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import preferencesReducer from 'calypso/state/preferences/reducer';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
 import { Banner } from '../index';
 
+jest.mock( 'calypso/my-sites/plan-price', () => ( { rawPrice, original } ) => (
+	<div
+		data-testid="banner-price-plan-mock"
+		className={ original ? 'is-original' : 'is-discounted' }
+	>
+		{ rawPrice }
+	</div>
+) );
+
+jest.mock( 'calypso/state/analytics/actions', () => ( {
+	recordTracksEvent: jest.fn( () => ( {
+		type: 'ANALYTICS_EVENT_RECORD',
```

Comment:
Is the returned action object necessary?

Reply:
I believe so. I paired with Noah on this one, but I recall we ran into an error without it. Do you have an alternative solution in mind?

Reply:
I think this is good already 👍 Was just generally curious.